File: TicTacToe.java
import java.util.Scanner;

public class TicTacToe 
{
    private static final char[][] board = new char[3][3];
    private static char currentPlayer = 'X';

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        initializeBoard();

        boolean gameOver = false;
        while (!gameOver) 
        {
            displayBoard();
            System.out.print("Player " + currentPlayer + ", enter your move (row [0-2] and column [0-2]): ");
            int row = input.nextInt();
            int column = input.nextInt();

            if (isValidMove(row, column)) 
            {
                board[row][column] = currentPlayer;

                if (checkWin(row, column)) 
                {
                    displayBoard();
                    System.out.println("Player " + currentPlayer + " wins!");
                    gameOver = true;
                } 
                else if (isBoardFull()) 
                {
                    displayBoard();
                    System.out.println("It's a draw!");
                    gameOver = true;
                } 
                else 
                {
                    currentPlayer = (currentPlayer == 'X') ? 'O' : 'X';
                }
            } 
            else 
            {
                System.out.println("Invalid move. Try again.");
            }
        }
    }

    private static void initializeBoard() 
    {
        for (int i = 0; i < 3; i++) 
        {
            for (int j = 0; j < 3; j++) 
            {
                board[i][j] = ' ';
            }
        }
    }

    private static void displayBoard() 
    {
        System.out.println("-------------");
        for (int i = 0; i < 3; i++) 
        {
            System.out.print("| ");
            for (int j = 0; j < 3; j++) 
            {
                System.out.print(board[i][j] + " | ");
            }
            System.out.println();
            System.out.println("-------------");
        }
    }

    private static boolean isValidMove(int row, int column) 
    {
        return row >= 0 && row < 3 && column >= 0 && column < 3 && board[row][column] == ' ';
    }

    private static boolean checkWin(int row, int column) 
    {
        if (board[row][0] == currentPlayer && board[row][1] == currentPlayer && board[row][2] == currentPlayer) 
        {
            return true;
        }
        if (board[0][column] == currentPlayer && board[1][column] == currentPlayer && board[2][column] == currentPlayer) 
        {
            return true;
        }
        if (row == column && board[0][0] == currentPlayer && board[1][1] == currentPlayer && board[2][2] == currentPlayer) 
        {
            return true;
        }
        if (row + column == 2 && board[0][2] == currentPlayer && board[1][1] == currentPlayer && board[2][0] == currentPlayer) 
        {
            return true;
        }
        return false;
    }

    private static boolean isBoardFull() 
    {
        for (int i = 0; i < 3; i++) 
        {
            for (int j = 0; j < 3; j++) 
            {
                if (board[i][j] == ' ') 
                {
                    return false;
                }
            }
        }
        return true;
    }
}

File: tictactoe.py
def print_board(board):
    for row in board:
        print(' | '.join(row))
        print('-' * 9)

def check_winner(board):
    for i in range(3):
        if board[i][0] == board[i][1] == board[i][2] != ' ':
            return board[i][0]
        if board[0][i] == board[1][i] == board[2][i] != ' ':
            return board[0][i]
    
    if board[0][0] == board[1][1] == board[2][2] != ' ':
        return board[0][0]
    if board[0][2] == board[1][1] == board[2][0] != ' ':
        return board[0][2]
    
    if all(cell != ' ' for row in board for cell in row):
        return 'Draw'
    
    return None

def tic_tac_toe():
    board = [[' ' for _ in range(3)] for _ in range(3)]
    current_player = 'X'
    
    while True:
        print_board(board)
        print(f"Player {current_player}'s turn")
        try:
            row = int(input("Enter row (0-2): "))
            col = int(input("Enter column (0-2): "))
            if board[row][col] != ' ':
                print("Cell already taken! Try again.")
                continue
            board[row][col] = current_player
        except (ValueError, IndexError):
            print("Invalid input! Please enter row and column as two numbers between 0 and 2.")
            continue
        
        winner = check_winner(board)
        if winner:
            print_board(board)
            if winner == 'Draw':
                print("It's a draw!")
            else:
                print(f"Player {winner} wins!")
            break
        
        current_player = 'O' if current_player == 'X' else 'X'

if __name__ == "__main__":
    tic_tac_toe()